src/components/views/context_menus/RoomGeneralContextMenu.tsx

```diff
@@ -213,18 +217,33 @@ export const RoomGeneralContextMenu: React.FC<RoomGeneralContextMenuProps> = ({
     }
 
     const { level } = useUnreadNotifications(room);
-    const markAsReadOption: JSX.Element | null =
-        level > NotificationLevel.None ? (
-            <IconizedContextMenuCheckbox
-                onClick={() => {
-                    clearRoomNotification(room, cli);
-                    onFinished?.();
-                }}
-                active={false}
-                label={_t("room|context_menu|mark_read")}
-                iconClassName="mx_RoomGeneralContextMenu_iconMarkAsRead"
-            />
-        ) : null;
+    const markAsReadOption: JSX.Element | null = (() => {
```

Comment:
this whole "define an anonymous function and immediately call it" seems like the worst sort of javascript braindamage to me. What's wrong with 

```
const markAsReadOption: JSX.Element | null = null;
if (level > NotificationLevel.None) {
    markAsReadOption = ...;
} else if (...) {
    markAsReadOption = ...;
}
```
?

(Also, call it `markAsReadOrUnreadOption` ?)

Reply:
Only because that const would then have to be a `let`, so lets us use const instead (it's replacing a ternary, I think, which of course allows the variable to be const, but the expression would have been a mess of nested ternaries). I can make it a let if that's preferable.